api/test/test_covjson.py

```diff
@@ -0,0 +1,104 @@
+# Test covjson.py
+import json
+
+import datastore_pb2 as dstore
+import pytest
+from covjson_pydantic.coverage import Coverage
+from covjson_pydantic.coverage import CoverageCollection
+from fastapi import HTTPException
+from formatters.covjson import Covjson
+from google.protobuf.json_format import Parse
+
+
+def test_single_parameter_convert():
+    test_data = load_json("test/test_data/test_single.json")
+    compare_data = load_json("test/test_data/test_single_compare.json")
+
+    response = create_mock_obs_response(test_data)
+    coverage_collection = Covjson().convert(response)
+
+    assert coverage_collection is not None
+
+    assert type(coverage_collection) is Coverage
+
+    # Assert that the coverage collection has the correct parameter
+    # TODO: Change parameter name when parameter names have been decided
+    assert "ff" in coverage_collection.parameters.keys()
+
+    # Check that correct values exist in the coverage collection
+    assert 9.21 in coverage_collection.ranges["ff"].values
+
+    # compare the coverage collection with the compare data
+    # TODO: Modify compare data when parameter names have been decided
+    coverage_collection_json = json.loads(coverage_collection.model_dump_json(exclude_none=True))
+    assert coverage_collection_json == compare_data
+
+
+def test_multiple_parameter_convert():
+    test_data = load_json("test/test_data/test_multiple.json")
```

Comment:
I would lower the number of parameters to 3 (what does more then that test), and the number of time points to 3 as well.
Just to lower test maintenance cost.

Reply:
fixed in https://github.com/EURODEO/e-soh/pull/24/commits/448804e7c55e356044cd83bc250ccf4bc3c433c0